indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -0,0 +1,150 @@
+                .join(RegistrationData.field_data)
+                .filter(RegistrationData.registration_id == Registration.id)
+                .filter(db.or_(*criteria))
+                .correlate(Registration)
+                .as_scalar())
+    return query.filter(subquery == len(filters))
+
+
+def _query_registrations(regform, filters):
+    query = (Registration.query
+                         .with_parent(regform)
+                         .options(joinedload('data').joinedload('field_data').joinedload('field')))
+    return _filter_registration(query, filters)
+
+
+class RHRegistrationsListManage(RHManageRegFormBase):
+    """List all registrations of a specific registration form of an event"""
+
+    def _process(self):
+        session_key = 'reg_list_config_{}'.format(self.regform.id)
+        report_config_uuid = request.args.get('config')
+        if report_config_uuid:
+            configuration = cache.get(report_config_uuid)
+            if session_key in session:
+                session[session_key] = configuration
+            else:
+                session.setdefault(session_key, configuration)
+            return redirect(url_for('.manage_reglist', self.regform))
+        reg_list_config = session.get(session_key, {'items': [], 'filters': {}})
+        regform_items = RegistrationFormItem.find_all(RegistrationFormItem.id.in_(reg_list_config['items']))
+        registrations = _query_registrations(self.regform, reg_list_config['filters']).all()
+        return WPManageRegistration.render_template('management/regform_reglist.html', self.event, regform=self.regform,
+                                                    event=self.event, visible_cols_regform_items=regform_items,
+                                                    registrations=registrations)
+
+
+class RHRegistrationsListCustomize(RHManageRegFormBase):
+    """Filter options and columns to display for a registrations list of an event"""
+
+    def _process_GET(self):
+        session_key = 'reg_list_config_{}'.format(self.regform.id)
+        reg_list_config = session.get(session_key, {'items': [], 'filters': {}})
+        filters = list(itertools.chain(*reg_list_config['filters'].itervalues()))
+        return WPManageRegistration.render_template('management/reglist_filter.html', self.event, regform=self.regform,
+                                                    event=self.event, RegistrationFormItemType=RegistrationFormItemType,
+                                                    visible_cols_regform_items=reg_list_config['items'],
+                                                    filters=filters)
+
+    def _process_POST(self):
+        filters = _get_filters_from_request()
+        session_key = 'reg_list_config_{}'.format(self.regform.id)
+        visible_regform_items = json.loads(request.values['visible_cols_regform_items'])
+        session.setdefault(session_key, {})
```

Comment:
you could assign the return value to a variable and then use that in the next two lines instead of `session[session_key]`